### ISSUE TITLE
can be windows OS with only CPU used ?

### ISSUE THREAD

**Issue by Sandy4321** [karpathy/ng-video-lecture#37]
can be windows OS with only CPU used ?

**Comment by fasterinnerlooper** [comment]
Yes, but it will take a very long time to complete one iteration.

**Comment by Sandy4321** [comment]
FOE EXAMPLE , 20 SECONDS?

**Comment by AdityaKulshrestha** [comment]
Longer than that, depends on your system hardware.

**Comment by gyllila** [comment]
I have rewritten gpt.py to run it on my laptop with CPU only. By keeping the numbers of layers, n_embd and block_size, I only managed to have 500 iters done per day (my laptop doesn’t work at night) and I don’t know if it works on PC.
I’m also not sure if I did it right because I don’t know much about Shakespeare. The output looks much like more.txt, but not much like input.txt. Here a sample:
The top in a world. You virt, the ban't,
Wallr most us, he hath as from all nock,
The cheke my son me queet, give me in!
For Juwe'le:
Be Lord Cancion just and kind.
If Petter for of death;
Slay of world he while not a beling the sheep;
The Warwick, the sweares pake shall not.

If after reading this, you still want to give it a try, here is the direct link to the Python file on GitHub:
https://github.com/gyllila/DemoLLMDownsizing/blob/main/gpt_embedding_only_s.py
